mne/viz/evoked.py

```diff
@@ -223,7 +223,9 @@ def _plot_evoked(evoked, picks, exclude, unit, show, ylim, proj, xlim, hline,
             raise ValueError('exclude has to be a list of channel names or '
                              '"bads"')
 
-        picks = list(set(picks).difference(exclude))
+        picks = [pick for pick in picks if pick not in exclude]
+    if len(picks) != len(set(picks)):
+        picks = set(picks)
```

Comment:
why not always do this? And it should maybe be `picks = sorted(set(picks))`

Reply:
The is that we can determine the order of channels via setting picks. This allows, eg, anterior/posterior sorting of channels. See the example PR.

Reply:
My more general point is that doing `set(...)` in one branch is dangerous, precisely because of this difference it can make in ordering:
```
>>> list(set([1, 5, 4, 4]))
[1, 4, 5]
```
So if what I suggested you change it to -- always explicitly force things to be ordered -- will not work for this use case, what you have here will likely lead to subtle bugs depending on the branching.

Reply:
Well, right now the ordering will be random if `exclude` is not an empty list, which the code as-is would change to ordering being random if `picks` is not unique.

One way to deal with this would be to warn or error if picks are not unique.

Another would be to warn + going explicitly:
```
new_picks = list()
for pick in picks:
    if pick not in new_picks:
        new_picks.append(pick)
picks = new_picks
```
and documenting this behaviour.

Reply:
Yes the old code has this ordering problem, too :)

How about we change it to be an error if it's not unique, and we can relax it if we hit a use case where it's good to allow repetitions?

Reply:
Makes sense to me.

Reply:
Also a good reason to finally force the full circle build.